CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -87,8 +87,27 @@ public class Child {
   /// This is used to e.g. describe the child if all of its tokens are missing in the source file.
   public let nameForDiagnostics: String?
 
-  /// A doc comment describing the child.
-  public let documentation: SwiftSyntax.Trivia
+  /// A summary of a doc comment describing the child. Full docc comment for the
+  /// child is available in ``Child/documentation``, and includes detailed list
+  /// of possible choices for the child if it's a token kind.
+  public let documentationSummary: SwiftSyntax.Trivia
```

Comment:
The good old `documentation` is now `documentationSummary`, but the spots that used it would benefit from having the list in them. That's why this commit does not need to change `SyntaxNodesFile.swift` at all — we just change the underlying `child.documentation`.